### ISSUE TITLE
Why did you change the Bookshelf sprite?

### ISSUE THREAD

**Issue by grandsong** [yooksi/pa-aio-deluxe#1]
I've noticed that you added a brand new set of images of Bookshelf in sprite.png.
The visual size of the new object is 1x2 while my old one is 1x1.
This is a very big change.
Since Bookshelf is highly commonly used across the game, I have to ask you about the reason for such a change.

And is it possible to treat your version of Bookshelf as an alternative of the existing one? For example, naming the former "Larger Bookshelf"?

**Comment by matshou** [comment]
One of my main gripes with the game is how ridiculously out of proportion many items are, and since as you say the bookshelf item is very commonly used I decided to re-scale it to be more realistic. It is now more wide along X-axis to better represent how a real shelf actually looks like (it is uncommon to find a shelf that holds only 3 books) and less wide along Y-axis so it no longer appears as it is blocking movement in cells and one tile corridors.

 > And is it possible to treat your version of Bookshelf as an alternative of the existing one?

 We could do that if you disagree with my interpretative vision of the bookshelf item, I would have no objections.

**Comment by grandsong** [comment]
I am not sure which version is more realistic, the compact one or the larger one.

As far as I know, there are no bookshelves at all in most prison cells.

My guess is that a bookshelf, if there is one, in a cell is not for a prisoner to read very much but just to store very few books (a bible, and one or two books borrowed from a library) or small personal paper things like photos, letters, cards, etc. 

It is a prison after all. Inmates could not be treated as nicely as college students.

If my guess is right, the vanilla version seems to make sense.

**Comment by matshou** [comment]
> My guess is that a bookshelf, if there is one, in a cell is not for a prisoner to read very much but just to store very few books (a bible, and one or two books borrowed from a library) or small personal paper things like photos, letters, cards, etc.

I disagree, here is an example of a bookshelf in a death-row cell: <sup>[^1]</sup>

![2020-12-25_023955](https://user-images.githubusercontent.com/16436920/103113328-835f6480-465a-11eb-981a-e25f14634780.png)

But in any case I think the best thing to do (as you suggested) is go with two separate versions of bookshelf, and let the player choose which one he wants to use. As is almost always the case the more variation the better.

[^1]: https://youtu.be/7tqypS2cm0g?t=1429

**Comment by grandsong** [comment]
I've separated the two versions. I renamed yours as BigBookshelf = "Big Bookshelf".
Feel free to rename it as you like.
I edited a few properties of "Big Bookshelf" like price.
I also edited the sprite so that it can align to bottom correctly.

Since I've checked all changes and modified what I think necessary, I pushed file to Git repo.

**Comment by grandsong** [comment]
A little reminder: you may want to add BigBookshelf to LikelyObject sections of some rooms like cell.

**Comment by matshou** [comment]
> Since I've checked all changes and modified what I think necessary, I pushed file to Git repo.

Could we have all the new changes pushed to a feature branch [v6.5](https://github.com/yooksi/pa-aio-deluxe/tree/v6.5) ?
That way when we are done we can do a clean merge to `master` and tag the merge commit as a release.

**Comment by grandsong** [comment]
Sorry but I am not very familar with features about branch. 
How to do that?

Moreover, why do we need to release in such a formal way?
The files in this repo are mostly just what the mod uses. If somebody want to download the mod from here instead of in Steam Workshop, they can just download the repo.
And I doubt anyone really have a need for an old version.

**Comment by matshou** [comment]
> Sorry but I am not very familar with features about branch.
> How to do that?

If you are using [Github Dektop](https://desktop.github.com/):

- Switch to branch `v6.5` (it will probably be called `origin/v6.5`)
- Make your changes as you would normally do.
- Switch to `Changes` tab in Github Desktop.
- Select your changes by ticking checkbox next to file entries in left panel.
- Write commit name and (optionally) description.
- Hit the blue button that says `Commit to v6.5`
- Click the `Push origin` button to push changes to Github.

For more information see the following images:
<ul>
<li>
<details>
<summary>How to switch to a different branch:</summary>
<img src="https://user-images.githubusercontent.com/16436920/103135677-21364c00-46ba-11eb-83d9-36c5810bb048.png">
</details>
</li>
<li>
<details>
<summary>How to stage files and create a commit:</summary>
<img src="https://user-images.githubusercontent.com/16436920/103135759-df59d580-46ba-11eb-8875-7ffe9d94cd01.png">
</details>
</li>
<li>
<details>
<summary>How to push commits to Github:</summary>
<img src="https://user-images.githubusercontent.com/16436920/103136162-4200a080-46be-11eb-8bca-a8d1ce6be657.png">
</details>
</li>
</ul>

If you are using Git via bash console or command prompt:
- Fetch repository changes:
`$ git fetch`
- Pull and checkout branch from Github: 
`$ git checkout v6.5`
- Make and push the commit as you would normally do.

> Moreover, why do we need to release in such a formal way?
> The files in this repo are mostly just what the mod uses. If somebody want to download the mod from here instead of in Steam > Workshop, they can just download the repo.
> And I doubt anyone really have a need for an old version.

We do not **need** to do this, but it makes maintaining the project **a lot easier**, especially when we are talking about hundreds of small changes over many iterative releases in a collaborative environment. 

Merge commits allow us to do the following:
- Quickly see our release changes.
- Quickly [`checkout`](https://git-scm.com/docs/git-checkout) different releases for testing purposes.
- Do a reliable [`diff`](https://git-scm.com/docs/git-diff) view to compare two releases.
- Make things organized and clean when viewing our repo tree using [gitk](https://git-scm.com/docs/gitk).

However beneficial features such as branches are I understand that many do not want to bother with version control complexities as it takes time and patience to maintain a proper version control tree. For this reason I am always prepared to do these things myself, so if you feel this way of doing things is a bit *too much* just push your commits in `master` branch and I will rebase them to a feature branch. 

Keep in mind that this way we always have two authors listed on *rebased* commits and you will have to re-clone the repository for each release as I have to constantly force push into master to remove the commits.

**Comment by grandsong** [comment]
I had already pushed to master before you asked me to push to v6.5. So now in my GitHub Desktop, there is nothing for me to push to v6.5...
Do I have to revert my commit first?

**Comment by grandsong** [comment]
Just now I reverted my commit and as a result I lost all my changes!
The local files in my PC of the repo became your old ones.

**Comment by matshou** [comment]
I have cleaned up the `master` with a force push and rebased your [changes](https://github.com/yooksi/pa-aio-deluxe/commit/51f89db196e2ab75bba6748c0d41d5c0fc7f38e4) to `v6.5`.
Backup any new changes you might have made since the new commit and re-clone the repository.
After this just copy your backed up files (if any) back to repository and make further new commits to branch `v6.5`.

**Comment by grandsong** [comment]
OK. I see what the branch v6.5 is now.
However, some files are bit older than my lastest, so I pushed another commit.

After I pushed the commit, GitHub Desktop says "Create a Pull Request from you current branch" ... "to propose and collaborate on your changes".
Should I do that?

**Comment by matshou** [comment]
> After I pushed the commit, GitHub Desktop says "Create a Pull Request from you current branch" ... "to propose and collaborate on your changes".
> Should I do that?

Not sure why it is asking you that, my guess is because you are a repository collaborator and it is reminding you that you can make a pull request (probably into `master`) since that is the standard way of contributing to a larger project.

Since it's just the two of us and we are working on a fairly straight forward thing there is no need for us to do pull requests. We do might end using them though, for example if you or me wanted to implement a change on `v6.5` that we were not sure the other one would like we would create a new branch, make a commit there and then do a pull request into `v6.5` and request a review where the other collaborator would write his thoughts (via comments on code lines or general PR comments). If they both agreed the changes were good they would merge the pull request into target branch, otherwise the pull request would be rejected.

**Comment by grandsong** [comment]
I see. 
So, there is nothing I should do now.

I'll wait for you to inform me when you complete all changes and are ready to publish them.

**Comment by matshou** [comment]
Okay, I am currently rushing to push a release in another project, but I will try to prioritize this after that.
